libnetwork/internal/kvstore/boltdb/boltdb.go

```diff
@@ -180,6 +180,27 @@ func (b *BoltDB) AtomicDelete(key string, previous *store.KVPair) error {
 	})
 }
 
+// DeleteIdempotent deletes a value at "key". Unlike AtomicDelete it
+// doesn't check whether the deleted key is at a specific version before
+// deleting.
+func (b *BoltDB) DeleteIdempotent(key string) error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	return b.client.Update(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(b.boltBucket)
+		if bucket == nil {
+			return store.ErrKeyNotFound
+		}
+
+		val := bucket.Get([]byte(key))
+		if val == nil {
+			return store.ErrKeyNotFound
+		}
```

Comment:
I think it's just naming ... but, to be idempotent (same result when deleting an object twice), should this ignore the object not existing? bucket.Delete() returns 'nil' if the object didn't exist, so perhaps it can just be called unconditionally?

Reply:
Oh! Missed this discussion as I was reviewing. Funny; I actually left a comment about the "idempotent" part.

I always considered "does not exist" to be the _idempotent_ part (i.e., you can call delete multiple times, and the effect is the same: "the object is gone". That means; if the object was already gone, it would still complete successfully).

But  see https://github.com/moby/moby/pull/47422#discussion_r1497228406 where I was thinking if "idempotent" is the actually correct terminology here, as it feels more like a "force delete" (ignoring any mutations on the object that may have happened concurrently).